src/tests/test_pyproxy.py

```diff
@@ -2096,3 +2096,78 @@ def test_pyproxy_of_list_fill(selenium, func):
     assert func(a) is a
     func(ajs)
     assert a == ajs.to_py()
+
+
+def test_pyproxy_instanceof_function(selenium):
+    weird_function_shim = ""
+    if selenium.browser in ["firefox", "node"]:
+        # A hack to make the test work: In node and firefox this test fails. But
+        # I can't reproduce the failure in a normal browser / outside of the
+        # test suite. The trouble seems to be that the value of
+        # `globalThis.Function` changes its identity from when we define
+        # `PyProxyFunction` to when we execute this test. So we store `Function`
+        # on `pyodide._api.tests` so we can retrieve the original value of it
+        # for the test. This is nonsense but because the failure only occurs in
+        # the test suite and not in real life I guess it's okay????
+        # Also, no clue how node and firefox are affected but not Chrome.
```

Comment:
@WebReflection does this make any sense to you why the Function prototype would change? If I store:
```js
pyodide._module.tests.Function = Function
```
and then later compare:
```js
console.log(pyodide._module.tests.Function === Function);
```
this logs `false`. This causes the test to fail because `pyFunc` is an instanceof `pyodide._module.tests.Function` but not of `Function`. But the tests also only fail in Firefox and node and not in chrome...

Reply:
is the prototype that changes or the global reference? the only thing I can think about in *NodeJS* is that the `.call` doesn't default to `globalThis` apparently, as mentioned in MDN, and maybe that runs before Firefox and it patches somehow the function globally to simulate a browser via node? Or maybe the test suite uses JSDOM and it changes somehow the global function? I don't have other clues but it's weird indeed.